### ISSUE TITLE
Dokumenthanterare: Koppling karta - kapitel fungerar inte m.m

### ISSUE THREAD

**Issue by ingvar-uddevalla** [hajkmap/Hajk#279]
**Så här är det:**
Dokumenthanterare (eller som vi i Uddevalla och i ÖP-sammanhang kallar den Kapitelindelning) är generellt sett bra och Kungsbacka m.fl har ju lagt till en utskriftsmöjlighet m.m som verkar lovande. Dock kvarstår en del irriterande buggar/avvikelser.
Kopplingen från karta till kapitel fungerar inte. Inga lagerkort (cards) visas här heller (jfr issue #276 för Sök).

![image](https://user-images.githubusercontent.com/54808837/64183156-b8461780-ce69-11e9-9d23-f5664b36b27e.png)

**Så här vill vi ha det:**
Vi vill kunna lägga en länk till relaterat kapitel i Dokumenthanteraren från Lagerhanteraren, från lagerkorten/cards (t ex genom ikonen ”T”) och om möjligt med en länk från attribut-popupen (infoklicken).
När man klickar på Karta och vill visa de lager som t ex representerar kapitlet Energiproduktion, så ska tidigare tända lager fortsatt vara tända. Annars förstår nog inte användarna varför de tidigare lända lagren släcktes. Bäst är förstås om man som administratör kan ställa in om hur denna rensning ska utföras (allt eller delar av).
Innehållsförteckningen behöver en logisk följd. Man ska kunna ta sig vidare (och tillbaka) utan att behöva gå tillbaka till hela innehållsförteckningen. Behövs ex. knapp till nästa kapitel.
![image](https://user-images.githubusercontent.com/54808837/64183309-fc391c80-ce69-11e9-9878-cfffa7a17372.png)

**Versioner:**
Hajk 3 Beta. Testad huvudsakligen i Microsoft Edge. Ej testad i mobiltelefon.
**Notera**
Denna "issue" är en av de "Buggar/synpunkter på ÖP-verktygen" i HAJK 3 Beta" beskrivna 30 aug 2019 av användare i Uddevalla kommun.

**Comment by victoraxbom** [comment]
Kopplingen mellan kapitel och karta fungerar här hos oss! Länken till de relaterade kapitlen dyker upp både på de små "lagerkorten" och i lagerhanteraren.

**Comment by victoraxbom** [comment]
@jacobwod Har testat mot master med konfigurationsfiler som fungerat tidigare men nu fungerar det inte längre, det kommer alltså inga `Breadcrumbs` när man trycker upp en karta från öp-fönstret. Inte heller kopplingen mellan kartlager och textkapitel verkar fungera som den ska. Kartan kraschar nu, det verkar inte som den hittar fönstret med öp-texten.

**Comment by jacobwod** [comment]
Bra att veta, misstänkte att det kunde ha hänt nåt. Det är bara att ta tag i det, men jag har dessvärre inga konfigurationsfiler som jag med säkerhet vet fungerar (har inte använt funktionen). 

Vet du om det var nåt speciellt man skulle slå på för att `Breadcrumbs` skulle komma? Jag har en enkel ÖP-fil som bara har en "Visa i karta" knapp, och när jag trycker på den så zoomar den mig till rätt läge och öppnar upp rätt lager. Däremot får inte heller jag några `Breadcrumbs` och därför funderar jag på om det var nåt mer man skulle göra eller om de kom automatiskt?

**Comment by victoraxbom** [comment]
Jag kan skicka över de konfigurationsfiler som fungerade här hos oss. Lagerhanteraren behövde vara satt som en widget för att `Breadcrumbs` skulle synas. Lite dumt, det borde vara en separat inställning i admin om man vill använda dem eller inte, helt frikopplat från widget/toolbar.

**Comment by jacobwod** [comment]
Just det, LayerSwitcher ska vara `target=left|right`. Men då vet jag att det var det som gjorde om de syns eller inte hos dig och testar. Tack för förslaget om filerna men när jag tänker på det så kommer det knappast fungera speciellt bra: er layers.json innehåller väl också en mängd referenser som står internt och jag kommer då ändå behöva göra om det mesta för att nåt ska visas.

**Comment by victoraxbom** [comment]
Ja det stämmer, de flesta är interna. Så nog bättre du skapar en egen

**Comment by victoraxbom** [comment]
Om du hinner när du ändå är inne i koden får du gärna kika på att lägga in en inställning i admin för dem.

**Comment by jacobwod** [comment]
Är aldrig inne i `mapservice` faktiskt. Skulle kunna göra en flagga i `new-client` som läses men nån som sitter med .NET får gärna ta och fixa så flaggen faktiskt skrivs i `mapservice`.

**Comment by victoraxbom** [comment]
Har testat senaste master och nu dyker tyvärr inte länkningen till kapitel upp alls, varken under lagret eller på `Breadcrumbs`. Men du kanske är inne och jobbar med det @jacobwod ?

**Comment by victoraxbom** [comment]
Det var configfilen som var korrupt, nu syns länkningen igen. Kraschar dock fortfarande när man försöker öppna kapitlet.

**Comment by jacobwod** [comment]
OK, kan du lämna nån skärmdump och markera i den var du klickar när kraschen inträffar? Får du Reacts felmeddelande så får du gärna klistra in en dump här också, så har jag lite mer att gå på. 👍

**Comment by victoraxbom** [comment]
Det är när man klickar på dessa:

![image](https://user-images.githubusercontent.com/3994163/65152647-a4cdab80-da28-11e9-97a8-376114b86d58.png)

![image](https://user-images.githubusercontent.com/3994163/65152730-cdee3c00-da28-11e9-9be4-2072e9f0cd8d.png)

Felet:

![image](https://user-images.githubusercontent.com/3994163/65152828-0130cb00-da29-11e9-8774-bcfd0a95db9c.png)

Sen är det motsvarande fel då i `Breadcrumb.js`. Misstänker att det har att göra med hur du har byggt om hanteringen av fönstren. Den verkar inte hitta `informative`-fönstret längre

**Comment by jacobwod** [comment]
Aja, det ska jag kunna lösa bara jag återskapar problemet här. Hur ställer du in för att få en sån knapp, kan du visa en dump från admin också?

**Comment by victoraxbom** [comment]
Toppen! Knapparna dyker upp bara man har en kartlager kopplad till ett kapitel i översiktsplanstexten. 

![image](https://user-images.githubusercontent.com/3994163/65153550-3db0f680-da2a-11e9-8c0e-4a4845d8ebe6.png)

**Comment by jacobwod** [comment]
@victoraxbom @ingvar-uddevalla Ska funka nu, återkoppla gärna om det är ok hos er med.

**Comment by jacobwod** [comment]
Helt rätt, funderat på det också men lät det vara i nuläget. Kanske bättre att ändå tvinga att Informative-fönstret öppnas automatiskt, så man ser att det händer saker. Ska kolla på att lägga in det.

**Comment by victoraxbom** [comment]
Kan vara bra att det öppnas automatiskt om det inte redan är öppet. Om du har tid får du gärna kolla på det

**Comment by ingvar-uddevalla** [comment]
Bra. Ska kolla detta imorgon. (Vi har ju en autoimatinstallation som körs varje natt). Jag har dock problem med att verifiera övriga buggar trots uppdaterade backend-lösningar m.m. TRor det ligger knepigheter i mina config-filer. T ex så går det fortfarande inte att spara Breadscrums-inställningarna m.m i oversiktsplan.json-filen i min miljö (se #276) och därmed kan jag inte testa detta heller.

**Comment by jacobwod** [comment]
> Kan vara bra att det öppnas automatiskt om det inte redan är öppet. Om du har tid får du gärna kolla på det

Precis, det är det som var kvar och nu är det fixat i och med kommande commit. Hold on.

**Comment by victoraxbom** [comment]
> > Kan vara bra att det öppnas automatiskt om det inte redan är öppet. Om du har tid får du gärna kolla på det
> 
> Precis, det är det som var kvar och nu är det fixat i och med kommande commit. Hold on.

Toppen! Jag har testat och det verkar fungera bra nu!

**Comment by jacobwod** [comment]
@victoraxbom Jag har ifs ett problem som återstår: när man kör via `npm start` så fungerar allt perfekt. Men när jag bygger ut med `npm run build` så får jag följande fel när jag klickar på "pilen" i Breadcrumbs som då ska öppna Informativeverktyget:
![image](https://user-images.githubusercontent.com/110222/66630708-16061600-ec05-11e9-99ca-926f6d37bd7e.png)

Kan du bygga ut hos dig och se om det fungerar eller är lika illa? Hoppas att det är nåt lokalt cacheproblem bara, men så irriterande att inte bli av med det...

**Comment by jacobwod** [comment]
OK, jag vet vad det är och är på väg att lösa det.

**Comment by jacobwod** [comment]
262a8f6dc7960b136a6110a646f2e6a569ec3ba0 closes #279.

**Comment by victoraxbom** [comment]
@jacobwod Bra att det löste sig, Kan bekräfta att det fungerar bra att bygga och köra nu.

**Comment by jacobwod** [comment]
Bra, tack för återkopplingen.
